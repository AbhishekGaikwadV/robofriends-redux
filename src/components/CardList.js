import React from 'react';
import Card from './Card';
import {connect} from 'react-redux';



const CardList = (props) => {
    const {robots, searchField} = props;

    const filteredRobots = robots.filter(robot => {
        return robot.name.toLowerCase().includes(searchField.toLowerCase());
    });

  return (
    <div>
      { filteredRobots.map(robot => (
          <Card key={robot.id} {...robot}  />) )
      }

    </div>
  );
};

const mapStateToProps = (state) => ({
    searchField: state.robots.searchField,
    robots: state.robots.robots,
});



export default connect(mapStateToProps, )(CardList);