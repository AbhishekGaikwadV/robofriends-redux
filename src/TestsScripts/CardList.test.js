import { shallow, mount, render } from 'enzyme';
import React from 'react';
import CardList from '../components/Card';

// npm test -- --coverage


describe("CardList", ()=>{
    it("CardList SnapShot", () => {
        const mockRobots =[
            {
                id: 1,
                name: "QQQ",
            }
        ];

        expect(shallow(<CardList robots={mockRobots} />)).toMatchSnapshot();
    });
} );

