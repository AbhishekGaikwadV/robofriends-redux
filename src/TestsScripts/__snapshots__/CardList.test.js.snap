// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Name 1`] = `ShallowWrapper {}`;

exports[`CardList CardList SnapShot 1`] = `ShallowWrapper {}`;
